packages/headless/src/features/search-parameters/search-parameter-serializer.ts

```diff
@@ -281,7 +285,10 @@ function isValidDateRangeValue(date: string) {
   }
 }
 
-export function buildDateRanges(ranges: string[]) {
+export function buildDateRanges(
+  ranges: string[],
+  state: FacetValueState = 'selected'
+) {
   return ranges
```

Comment:
golly. map->filter->map. might be worth reworking this part to avoid that many array iterations (out-of-scope/nit-pik/scout camp)

Reply:
(also refactored buildNumericRanges)